dwave/system/utilities.py

```diff
@@ -17,10 +17,37 @@
 import os
 import json
 import networkx as nx
+import numpy as np
 import warnings
 
-__all__ = ['common_working_graph', 'classproperty']
+from typing import List, Union
 
+__all__ = [
+    'common_working_graph',
+    'classproperty',
+    'anneal_schedule_with_offset',
+    ]
+
+def _require(
+    argname: str,
+    array_like: np.typing.ArrayLike,
+    num_columns: int = 1,
+    ) -> np.ndarray:
+    "Coerce array-like input into a NumPy array."
+    try:
+        array = np.asarray_chkfinite(array_like)
+    except (ValueError, TypeError) as err:
+        raise ValueError(f"`{argname}`: {err}") from err
+
+    if not np.issubdtype(array.dtype, np.number):
+        raise TypeError(f"`{argname}` must be an array-like of numbers")
+
+    if array.ndim > 1 and array.shape[1] != num_columns or \
+        array.ndim == 1 and array.ndim != num_columns:
```

Comment:
pep8 recommends parenthesis over backspace for multiline ifs, and also operators after line break. E.g.
```suggestion
    if (array.ndim > 1 and array.shape[1] != num_columns
        or array.ndim == 1 and array.ndim != num_columns):
```